builder/builder.go

```diff
@@ -443,3 +453,88 @@ func parseImageNameFromArgs(cmdArgs string) string {
+			return err
+		}
+		return b.populateSecretVolume(ctx, volMount)
+	default:
+		return errors.New("volume source type not supported yet")
+	}
+}
+
+// createSecretFiles creates necessary files for source type Secret
+func (b *Builder) createSecretFiles(ctx context.Context, volMount *volume.Volume) error {
+	var args []string
+	var sb strings.Builder
+	if runtime.GOOS == util.WindowsOS {
+		args = []string{"powershell.exe", "-Command"}
+		sb.WriteString("mkdir " + volMount.Name)
+		log.Println("making directory " + volMount.Name)
+	} else {
+		args = []string{"/bin/sh", "-c"}
+		sb.WriteString("mkdir " + volMount.Name + " && ")
+	}
+	for k, v := range volMount.Source.Secret {
+		val := v
+		decoded, err := base64.StdEncoding.DecodeString(val)
+		if err != nil {
+			return errors.New("failed to decode Base64 value. please make sure value provided is Base64 encoded")
+		}
+		val = string(decoded)
+		if runtime.GOOS == util.WindowsOS {
+			sb.WriteString("; Add-Content -Path ")
+			sb.WriteString(volMount.Name + "/" + k)
+			sb.WriteString(" -Value @\"\n")
+			sb.WriteString(val)
+			sb.WriteString("\n\"@")
+		} else {
+			sb.WriteString("cat >> ")
+			sb.WriteString(volMount.Name + "/" + k)
+			sb.WriteString(" <<EOL\n")
+			sb.WriteString(val)
+			sb.WriteString("\nEOL\n")
+		}
+	}
+	args = append(args, sb.String())
+	var buf bytes.Buffer
+	if err := b.procManager.Run(ctx, args, nil, &buf, &buf, ""); err != nil {
+		return errors.Wrapf(err, "failed to write value, %s", buf.String())
+	}
+	log.Println("buffer: ", buf.String())
+	log.Println("Created new file(s) for volume: ", volMount.Name)
+	return nil
+}
+
+// populateSecretVolume mounts all files of type source Secret generated into a volume
+func (b *Builder) populateSecretVolume(ctx context.Context, volMount *volume.Volume) error {
+	var dataContainerArgs []string
+	var dataSB strings.Builder
+	if runtime.GOOS == util.WindowsOS {
+		dataContainerArgs = []string{"powershell.exe", "-Command"}
+		dataSB.WriteString("docker run --rm -v " + b.workspaceDir + ":c:\\source -v ")
+		dataSB.WriteString(volMount.Name + ":c:\\dest -w c:\\source ")
+		dataSB.WriteString(configImageName + " cmd.exe /c copy c:\\source\\" + volMount.Name + " c:\\dest")
+	} else {
+		dataContainerArgs = []string{"/bin/sh", "-c"}
+		dataSB.WriteString("docker run --rm -v " + b.workspaceDir + ":/source -v ")
+		dataSB.WriteString(volMount.Name + ":/dest -w /source " + configImageName + " cp ")
+		for k := range volMount.Source.Secret {
+			dataSB.WriteString(volMount.Name + "/" + k)
+			dataSB.WriteString(" ")
+		}
+		dataSB.WriteString("/dest")
+	}
+	dataContainerArgs = append(dataContainerArgs, dataSB.String())
+	var buf bytes.Buffer
+	if err := b.procManager.Run(ctx, dataContainerArgs, nil, &buf, &buf, ""); err != nil {
+		return errors.Wrapf(err, "failed to populate container, %s", buf.String())
+	}
+	log.Println("buffer: ", buf.String())
+	log.Println("Populated files for volume: ", volMount.Name)
```

Comment:
I guess instead of "populated" can we use something more generic like:

"Volume source `volMount.Name` successfully attached/mounted"